tests/xraywatch_test.go

```diff
@@ -0,0 +1,579 @@
+		return nil
+	}
+
+	artDetails := GetRtDetails()
+	artHTTPDetails := artDetails.CreateHttpClientDetails()
+	client, err := httpclient.ClientBuilder().Build()
+	if err != nil {
+		return nil
+	}
+
+	resp, _, err := client.SendDelete(artDetails.GetUrl()+"api/build/"+buildName+"?deleteAll=1", nil, artHTTPDetails)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return errors.New("failed unmarshalling build " + resp.Status)
+	}
+
+	return nil
+}
+
+type ArtifactoryBuild struct {
+	Version    string              `json:"version"`
+	Name       string              `json:"name"`
+	Number     string              `json:"number"`
+	Started    string              `json:"started"`
+	Properties interface{}         `json:"properties"`
+	Modules    []ArtifactoryModule `json:"modules"`
+}
+
+type ArtifactoryModule struct {
+	ID        string                `json:"id"`
+	Artifacts []ArtifactoryArtifact `json:"artifacts"`
+}
+
+type ArtifactoryArtifact struct {
+	Type string `json:"type"`
+	Sha1 string `json:"sha1"`
+	Md5  string `json:"md5"`
+	Name string `json:"name"`
+}
+
+func createPolicy(policyName string) error {
+	xrayDetails := GetXrayDetails()
+	xrayHTTPDetails := xrayDetails.CreateHttpClientDetails()
+
+	artUtils.SetContentType("application/json", &xrayHTTPDetails.Headers)
+	client, err := httpclient.ClientBuilder().Build()
+	if err != nil {
+		return err
+	}
+	data := ArtifactoryPolicy{
+		Name:        policyName,
+		Description: "example policy",
+		Type:        "security",
+		Rules: []ArtifactoryPolicyRules{{
+			Name:     "sec_rule",
+			Priority: 1,
+			Criteria: map[string]string{
+				"min_severity": "medium",
+			},
+			Actions: ArtifactoryPolicyActions{
+				Webhooks: []string{},
+				BlockDownload: ArtifactoryPolicyActionsBlockDownload{
+					Active:    true,
+					Unscanned: false,
+				},
+				BlockReleaseBundleDistribution: true,
+				FailBuild:                      true,
+				NotifyDeployer:                 true,
+				NotifyWatchRecipients:          true,
+			},
+		}},
+	}
+
+	requestContent, err := json.Marshal(data)
+	if err != nil {
+		return errors.New("failed marshalling policy " + policyName)
+	}
+
+	resp, _, err := client.SendPost(xrayDetails.GetUrl()+"api/v2/policies", requestContent, xrayHTTPDetails)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		return errors.New("Status is not Created - " + strconv.Itoa(resp.StatusCode))
+	}
+
+	return nil
+}
+
+func deletePolicy(policyName string) error {
+	xrayDetails := GetXrayDetails()
+	xrayHTTPDetails := xrayDetails.CreateHttpClientDetails()
+	artUtils.SetContentType("application/json", &xrayHTTPDetails.Headers)
+	client, err := httpclient.ClientBuilder().Build()
+	if err != nil {
+		return nil
+	}
+
+	resp, _, err := client.SendDelete(xrayDetails.GetUrl()+"api/v2/policies/"+policyName, nil, xrayHTTPDetails)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return errors.New("failed to delete policy " + resp.Status)
+	}
```

Comment:
I suggest changing the above to:
```
	if err != nil  {
		return error
	}
	if resp.StatusCode != http.StatusOK {
		return errors.New("failed to delete policy " + resp.Status)
	}
```
This also applies to other checks in this file.